examples/js/utils/SceneUtils.js

```diff
@@ -37,6 +37,52 @@
 
 	}
 
+	//Convert multi-material mesh into group of mesh with single material
+	function createMeshesFromMultiMaterialMesh( mesh ) {
+
+		const group = new THREE.Group();
+
+		//split groups by material index
+		const materials = mesh.material;
+
+		if ( ! ( materials instanceof Array ) ) {
+
+			console.error( 'Not a multimaterial mesh.' );
+			return;
+
+		}
+
+		materials.forEach( ( mat, index ) => {
+
+			const singleIndexGroups = mesh.geometry.groups.filter( g => g.materialIndex == index );
+			const singleIndices = mesh.geometry.index.array.filter( ( val, i ) => {
+
+				let belong = false;
+				singleIndexGroups.forEach( g => {
+
+					belong |= ( g.start <= i && g.count + g.start > i );
+
+				} );
+
+				return belong;
+
+			} );
+			const singleGeo = mesh.geometry.clone();
```

Comment:
It's probably better to write the method differently. Meaning instead of cloning the geometry over and over again, you create for each new mesh a new geometry but only with the buffer attribute data according to the respective group. This would be the more memory efficient solution.